carbon-design-system/carbon-addons-iot-react
# fix(datetimepicker): listen to locale and initialize dayjs with it

Closes #

**Summary**

- Fixes a bug where the relative dates were not taking locale into account.  I should be able to pass `L` as the dateTimeMask property to the DateTimePicker component and dayjs needs to have the correct locale set.

Notice that the date format here is incorrect for French
![image](https://user-images.githubusercontent.com/6663002/122832189-3fb38380-d2a8-11eb-97df-3bcdd4ce1b5e.png)

However, the date inside the Calendar widget is showing the French dates correctly

![image](https://user-images.githubusercontent.com/6663002/122832269-5954cb00-d2a8-11eb-9e1a-b57b838cb72a.png)


**Change List (commits, features, bugs, etc)**

- fix(DateTimePicker): make a story that uses non-default locales and set the locale inside dayjs
- chore(snaps): I have no idea why those two snaps were removed for List and Card, I don't see those stories anymore, but I didn't change them. I believe it was caused by this commit: https://github.com/carbon-design-system/carbon-addons-iot-react/commit/26c5ab798558b9f01a00669836f0e08370585a7d

**Acceptance Test (how to verify the PR)**

- Verify the locale story, if you popup the date selector you should see the Day then Month as the French date, and the Calendar should still show the French dates

**Regression Test (how to make sure this PR doesn't break old functionality)**

- Verify default locale is used in other places
